scripts/lib-carousel.js

```diff
@@ -0,0 +1,389 @@
+/**
+ * Slider duration in milliseconds for auto-advance.
+ * @type {number}
+ */
+const sliderDurationMs = 3500;
+
+// HTML code for arrow icons
+const HTML_ARROW_LEFT = '<svg fill="rgb(217, 217, 217)" xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 600 600">\n'
+  + '<path d="M97.141,225.92c0-8.095,3.091-16.192,9.259-22.366L300.689,9.27c12.359-12.359,32.397-12.359,44.751,0c12.354,12.354,12.354,32.388,0,44.748L173.525,225.92l171.903,171.909c12.354,12.354,12.354,32.391,0,44.744c-12.354,12.365-32.386,12.365-44.745,0l-194.29-194.281C100.226,242.115,97.141,234.018,97.141,225.92z"/>\n'
+  + '</svg>';
+const HTML_ARROW_RIGHT = '<svg fill="rgb(217, 217, 217)" xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 600 600">\n'
+  + '<path d="M345.441,248.292L151.154,442.573c-12.359,12.365-32.397,12.365-44.75,0c-12.354-12.354-12.354-32.391,0-44.744L278.318,225.92L106.409,54.017c-12.354-12.359-12.354-32.394,0-44.748c12.354-12.359,32.391-12.359,44.75,0l194.287,194.284c6.177,6.18,9.262,14.271,9.262,22.366C354.708,234.018,351.617,242.115,345.441,248.292z"/>\n'
+  + '</svg>\n';
+
+let activeSlide = 1;
+let slideShow = false;
+let slideCount = 0;
+let touchStartX = 0;
+let touchEndX = 0;
+let touchRelX = 0;
+
```

Comment:
Since you're using many DOM elements multiple times, it is beneficial to performance to cache them. Looks like you can cache the following elements:

```
const sliderWrapper = document.querySelector('.slider-wrapper');
const btnNav = document.querySelector('.dotted-nav');
const arrowNav = document.querySelector('.arrow-nav');
const slider = document.querySelector('.slider');
let btnNavChildren, sliderChildren;
```

Just below, you can fill the cache:
```
function updateCaches() {
  btnNavChildren = Array.from(btnNav.children);
  sliderChildren = Array.from(slider.children);
}

updateCaches();
```